objcgen/objcgenerator.cs

```diff
@@ -799,8 +799,127 @@ protected override void Generate (ProcessedMethod method)
+				break;
+			case TypeCode.Single:
+				implementation.WriteLine ("float __resarrval = mono_array_get (__resarr, float, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithFloat:__resarrval];");
+				break;
+			case TypeCode.SByte:
+				implementation.WriteLine ("signed char __resarrval = mono_array_get (__resarr, signed char, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithChar:__resarrval];");
+				break;
+			case TypeCode.Int16:
+				implementation.WriteLine ("short __resarrval = mono_array_get (__resarr, short, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithShort:__resarrval];");
+				break;
+			case TypeCode.Int32:
+				implementation.WriteLine ("int __resarrval = mono_array_get (__resarr, int, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithInt:__resarrval];");
+				break;
+			case TypeCode.Int64:
+				implementation.WriteLine ("long long __resarrval = mono_array_get (__resarr, long long, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithLongLong:__resarrval];");
+				break;
+			case TypeCode.UInt16:
+				implementation.WriteLine ("unsigned short __resarrval = mono_array_get (__resarr, unsigned short, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithUnsignedShort:__resarrval];");
+				break;
+			case TypeCode.UInt32:
+				implementation.WriteLine ("unsigned int __resarrval = mono_array_get (__resarr, unsigned int, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithUnsignedInt:__resarrval];");
+				break;
+			case TypeCode.UInt64:
+				implementation.WriteLine ("unsigned long long __resarrval = mono_array_get (__resarr, unsigned long long, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithUnsignedLongLong:__resarrval];");
+				break;
+			case TypeCode.Byte:
+				implementation.WriteLine ("NSData *__resobj = [NSData dataWithBytes:mono_array_addr (__resarr, unsigned char, 0) length:__resarrlength];");
+				break;
+			case TypeCode.String:
+				implementation.WriteLine ("MonoString* __resarrval = mono_array_get (__resarr, MonoString *, __residx);");
+				implementation.WriteLine ("if (__resarrval)");
+				implementation.Indent++;
+				implementation.WriteLine ("__resobj = mono_embeddinator_get_nsstring (__resarrval);");
+				implementation.Indent--;
+				implementation.WriteLine ("else");
+				implementation.Indent++;
+				implementation.WriteLine ("__resobj = [NSNull null];");
+				implementation.Indent--;
+				break;
+			case TypeCode.Object:
+				var tname = NameGenerator.GetTypeName (t);
+				if (types.HasProtocol (t))
+					tname = "__" + tname + "Wrapper";
+				implementation.WriteLine ("MonoObject* __resarrval = mono_array_get (__resarr, MonoObject *, __residx);");
+				implementation.WriteLine ($"{tname}* __tmpobj = [[{tname} alloc] initForSuper];");
+				implementation.WriteLine ("if (__resarrval) {");
```

Comment:
You're creating `__tmpobj` even if `__resarrval` is NULL, you can move the `__tmpobj` creation inside the `if` to avoid this.